### PR TITLE
(feat) add support to search relationship by patient identifier on registration form

### PR DESCRIPTION
**Pull request by Herobiam**
## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

Enhancing the registration form's relationship creation feature has been addressed in this pull request (PR). Formerly, the search functionality was limited to first and last names, omitting the option to search using the patient identifier. The proposed changes in this PR extend the search capabilities to encompass patient identifiers, thereby providing a more comprehensive and efficient method for establishing relationships within the registration process.

## Screenshots

https://github.com/openmrs/openmrs-esm-patient-management/assets/19629227/6da3f20b-3830-4793-8246-5e3869eb6940

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

File: packages/esm-patient-registration-app/src/patient-registration/patient-registration.resource.tsx [hunk 1]
```diff
@@ -182,11 +182,20 @@ export function usePatientPhoto(patientUuid: string): UsePatientPhotoResult {
...
-    signal: abortController.signal,
-  });
+  const BASE_API = '/ws/rest/v1';
+  const ctrl = new AbortController();
+  const { signal } = ctrl;
+
+  try {
+    const endpoint = (await openmrsFetch(`${BASE_API}/person?q=${query}`, { signal })).data.results.length
+      ? `${BASE_API}/person?q=${query}`
+      : `${BASE_API}/patient?q=${query}`;
```
Comment (pirupius) [inline packages/esm-patient-registration-app/src/patient-registration/patient-registration.resource.tsx:192]: This looks like something that could be optimized. 2 api calls will end up being made to `/person` with the second one being made  on line 194

File: packages/esm-patient-registration-app/src/patient-registration/patient-registration.resource.tsx [hunk 2]
```diff
@@ -182,11 +182,25 @@ export function usePatientPhoto(patientUuid: string): UsePatientPhotoResult {
...
+    } = await openmrsFetch(personEndpoint, { signal });
+
+    if (results.length > 0) {
+      return { data: { results } };
+    }
+
+    const patientEndpoint = `${BASE_API}/patient?q=${query}`;
+    return openmrsFetch(patientEndpoint, { signal });
+  } catch (error) {
+    console.error('Error fetching person:', error);
```
Comment (eudson) [inline packages/esm-patient-registration-app/src/patient-registration/patient-registration.resource.tsx:201]: @Herobiam can you remove this console log please

**Review by eudson — CHANGES_REQUESTED** [review]

**Comment by ibacher** [comment]
I’m closing this in favour of #856, which implements the same logic.

**Comment by Herobiam** [comment]
Okay great thanks @ibacher

**Pull request closed without merging**